### PR TITLE
New widget

### PR DESCRIPTION
**Pull request by janhohner**
As discussed in #148 and Gitter I moved the code for the new widget to a seperate file. This provides solutions to #135 and #142 (I don't know if I should mark them as solved because these issues still exist in the original widget implementation).

This version of the widget currently only supports evergreen browsers and IE9+.

### CODE REVIEW

**Comment by acabunoc** [comment]
Thanks for the new PR! I'll take a closer look this afternoon :)

File: README.md [hunk 1]
```diff
@@ -14,48 +14,7 @@ The PaperBadger widget enables *anyone*, from publishers to individual researche
...
-       document.write(decodeURIComponent("%3Cscript src='https://badges.mozillascience.org/widgets/paper-badger-widget.js' type='text/javascript'%3E%3C/script%3E"));
-    </script>
-    <script>
-        var conf={"article-doi": "10.1186/2047-217X-3-18", "container-class": "my-container"};
-        showBadges(conf);
-    </script>
-    </body>
-    </html>
-```
+You can either use paper-badger-widget.js ([documentation](docs/paper-badger-widget.md)), which supports old browsers or widget.js ([documentation](docs/widget.md)), which is in development and currently only supports evergreen browsers and IE9+.
```
Comment (acabunoc) [inline README.md]: I like how you handled this! Thanks.

We may want to start a wiki and move those docs there, but in the meantime this is great.

File: public/widgets/widget.js [hunk 2]
```diff
@@ -0,0 +1,355 @@
+/* exported PaperBadgerWidget */
+/* global ActiveXObject: false */
+/**
+ * PaperBadger widget
+ * Displays all badges for a DOI or ORCID.
+ *
+ * @param {Object}    settings
```
Comment (acabunoc) [inline public/widgets/widget.js]: Yay, thanks for the documentation here!

File: public/widgets/widget.js [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+   * @type {Number}
+   */
+  var loaderInterval;
+
+  /**
+   * Checks the settings given to the function and starts
+   * the widget creation.
+   */
+  var construct = function () {
+    // fix for #135, all but the first forward slash of a DOI
```
Comment (acabunoc) [inline public/widgets/widget.js]: Long term, we should probably handle this within the API, but this works great for now!

Comment (acabunoc) [inline public/widgets/widget.js]: Re: https://github.com/mozillascience/paperbadger/issues/135#issuecomment-144097283

File: public/widgets/widget.js [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+   * and starts an animation loop.
+   */
+  var startLoader = function () {
+    loaderElement = document.createElement('div');
+    loaderElement.className = 'paperbadger-loader';
+    loaderElement.setAttribute('data-step', '4');
+    containerElement.appendChild(loaderElement);
+    loaderMethod();
+
+    loaderInterval = setInterval(loaderMethod, 750);
```
Comment (acabunoc) [inline public/widgets/widget.js]: clever animation!

File: docs/widget.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+## Usage
+1. To use the widget on your own site, include a `<div>` with your custom class in your view file, for example:
+    `<div class="my-container"></div>`
+
+2. Above the closing `<body>` tag, add
+  ```html
+      <script>
+        (function () {
+          var d=document,g=d.createElement("script");g.type="text/javascript";g.async=!0;g.defer=!0;
+          g.src="https://badges.mozillascience.org/widgets/paper-badger-widget.js";g.onload=load;d.body.appendChild(g);
```
Comment (acabunoc) [inline docs/widget.md]: Should this be `https://badges.mozillascience.org/widgets/widget.js`?

File: docs/widget.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  * the `removeClass` key: the widget will remove this class from all elements once it has loaded, if this is a non-empty string
+  * the `clickCallback` key: a method that gets called when a link under a badge gets clicked. The method is called with an object containing either the DOI or ORCID of the link and the badge taxonomy:
+    
+    ```
+        {
+            doi: '10.1186/2047-217X-3-18', // either DOI
+            orcid: '0000-0001-5207-5061', // or ORCID
+            taxonomy: 'data_curation'
+        }
+    ```
```
Comment (acabunoc) [inline docs/widget.md]: Can you link to the demos you made here somewhere?

**Comment by acabunoc** [comment]
Looks great -- there's a couple comments I left in `docs/widget.md` that need to be looked at. Once that's clear, I think this is good to! :+1:

File: public/widgets/widget.js (hunk 4 above)
Comment (janhohner) [inline public/widgets/widget.js]: Thanks! I first thought I add a base64 encoded gif to the source, but then thought about file size and performance, so I did it that way.

File: docs/widget.md (hunk 5 above)
Comment (janhohner) [inline docs/widget.md]: You are right, I'll fix it.

File: docs/widget.md (hunk 6 above)
Comment (janhohner) [inline docs/widget.md]: I just pushed a new commit and added a seperate heading for demos. Is it ok like that?

**Comment by janhohner** [comment]
Thanks for the review! I changed the docs in the necessary places.

**Comment by acabunoc** [comment]
Looks good to me! Merging

**Pull request merged**